packages/slate/src/interfaces/node.ts

```diff
@@ -1,14 +1,93 @@
 import { produce } from 'immer'
-import { Editor, Element, ElementEntry, Path, Range, Text } from '..'
+import { Editor, Path, Range, Text } from '..'
+import { Element, ElementEntry } from './element'
+import { ExtendedType } from './custom-types'
 
 /**
  * The `Node` union type represents all of the different types of nodes that
  * occur in a Slate document tree.
  */
 
-export type Node = Editor | Element | Text
+export type BaseNode = Editor | Element | Text
+export type Node = ExtendedType<'Node', BaseNode>
```

Comment:
What does this `ExtendedType` helper actually do?

Reply:
It facilitates [module augmentation](https://www.typescriptlang.org/docs/handbook/declaration-merging.html#module-augmentation) in Slate so Slate users can specify their own types for `Element`, `Text`, etc. so long as the type extends Slate's base types for those nodes.  Pretty useful so we can get TS compile errors if nodes don't meet a specific type structure defined by the user.